### PR TITLE
[cling-cpt] Got rid of 4 global variables and put them into fetch and compile functions [skip-ci]

### PR DESCRIPTION
**Pull request by saisoma123**
# This Pull request: Reduces global variable mutation, making it easier to debug.

## Checklist:

- [X] tested changes locally
- [NA] updated the docs (if necessary)

This PR fixes #406 mentioned in meta-issue list (https://github.com/root-project/cling/issues/406)

### CODE REVIEW

File: interpreter/cling/tools/packaging/cpt.py [hunk 1]
```diff
@@ -542,6 +549,8 @@ def compile(arg):
     global prefix, EXTRA_CMAKE_FLAGS
     prefix = arg
 
+    EXTRA_CMAKE_FLAGS = args.get('with_cmake_flags')
```
Comment (jalopezg-r00t) [inline interpreter/cling/tools/packaging/cpt.py:552]: Maybe get rid of `EXTRA_CMAKE_FLAGS` in line 549 (introduced as `global`)?

File: interpreter/cling/tools/packaging/cpt.py [hunk 2]
```diff
@@ -590,6 +599,8 @@ def compile_for_binary(arg):
...
```
Comment (jalopezg-r00t) [inline interpreter/cling/tools/packaging/cpt.py:602]: Same here.

**Review by jalopezg-r00t — COMMENTED** [review]
I added a couple of comments.  I leave the approval to @vgvassilev, but I'm not sure about the change itself, i.e. I would not frown upon leaving them as globals.

File: interpreter/cling/tools/packaging/cpt.py (hunk 2 above)
Comment (saisoma123) [reply interpreter/cling/tools/packaging/cpt.py:602]: I'll remove that that thank you for letting me know

Comment (saisoma123) [reply interpreter/cling/tools/packaging/cpt.py:602]: Yeah just removed both references

**Comment by vgvassilev** [comment]
I am realizing that making removing the global variables which are meant to hold the values from the argument parser is probably not a good idea after all. It makes the code less clear for me. Is there a way to make a global variable in python read only?

I know we merged already at least one PR of that kind but I fear we might need to revert it... @saisoma123, @jalopezg-r00t, what do you think?

**Comment by saisoma123** [comment]
Hi @vgvassilev that last PR you merged removed the CPT_SRC_DIR and the REVISION, both of which don't have to do with the argument parser, so I think that one should be fine.  Yeah I see what you mean with this PR though, I think it'd make more sense to just keep it the way it was.

**Comment by vgvassilev** [comment]
@saisoma123, if you agree, let's then close this PR.

**Comment by saisoma123** [comment]
@vgvassilev Yeah I'll close it right now

**Pull request closed without merging**